snapshot/src/test/java/io/zeebe/snapshots/impl/FileBasedReceivedSnapshotTest.java

```diff
@@ -9,630 +9,276 @@
+    receiveSnapshot(firstPersistedSnapshot).persist().join();
 
     // when
-    final var committedSnapshot = takeAndReceiveSnapshot(index + 1, term).persist().join();
-    final var committedSnapshotFiles = committedSnapshot.getPath().toFile().list();
+    final PersistedSnapshot secondPersistedSnapshot = takePersistedSnapshot(2L);
+    final var secondReceivedPersistedSnapshot =
+        receiveSnapshot(secondPersistedSnapshot).persist().join();
 
     // then
-    assertThat(receiverPendingSnapshotsDir.toFile()).isEmptyDirectory();
-
-    final var snapshotDirs = receiverSnapshotsDir.toFile().listFiles();
-    assertThat(snapshotDirs).hasSize(1);
-
-    final var committedSnapshotDir = snapshotDirs[0];
-    assertThat(FileBasedSnapshotMetadata.ofFileName(committedSnapshotDir.getName()))
-        .map(FileBasedSnapshotMetadata::getIndex)
-        .hasValue(2L);
-    assertThat(committedSnapshotDir.listFiles())
-        .extracting(File::getName)
-        .containsExactlyInAnyOrder(committedSnapshotFiles);
+    assertThat(receiverSnapshotsDir)
+        .as("there is only the latest snapshot in the receiver's snapshot directory")
+        .isDirectoryContaining(p -> p.equals(secondReceivedPersistedSnapshot.getPath()));
   }
 
   @Test
   public void shouldNotRemovePendingSnapshotOnCommittingSnapshotWhenHigher() throws Exception {
     // given
-    final var olderPersistedSnapshot = takeSnapshot(1L, 0L);
+    final var olderPersistedSnapshot = takePersistedSnapshot(1L);
     final ReceivedSnapshot olderReceivedSnapshot = receiveSnapshot(olderPersistedSnapshot);
-    final var newPersistedSnapshot = takeSnapshot(2L, 0L);
+    final var newPersistedSnapshot = takePersistedSnapshot(2L);
 
     // when
-    receiveSnapshot(newPersistedSnapshot);
+    final ReceivedSnapshot receivedSnapshot = receiveSnapshot(newPersistedSnapshot);
     olderReceivedSnapshot.persist().join();
 
     // then
-    final var pendingSnapshotDirs = receiverPendingSnapshotsDir.toFile().listFiles();
-    assertThat(pendingSnapshotDirs).hasSize(1);
-
-    final var pendingSnapshotDir = pendingSnapshotDirs[0];
-    assertThat(FileBasedSnapshotMetadata.ofFileName(pendingSnapshotDir.getName()))
-        .map(FileBasedSnapshotMetadata::getIndex)
-        .hasValue(2L);
-    assertThat(pendingSnapshotDir.listFiles())
-        .extracting(File::getName)
-        .containsExactlyInAnyOrder(newPersistedSnapshot.getPath().toFile().list());
+    assertThat(receiverPendingDir)
+        .as(
+            "the latest pending snapshot should not be deleted because it is newer than the persisted one")
+        .isDirectoryContaining(p -> p.equals(receivedSnapshot.getPath()));
   }
 
   @Test
   public void shouldNotifyListenersOnNewSnapshot() throws Exception {
     // given
-    final var listener = mock(PersistedSnapshotListener.class);
-    final var index = 1L;
-    final var term = 0L;
+    final AtomicReference<PersistedSnapshot> snapshotRef = new AtomicReference<>();
```

Comment:
This and the following test can go to the interface tests.

Reply:
Same as above, I think it might make more sense to move them when you're refactoring the snapshot store (and impl) tests.